web/src/refresh-components/onboarding/useOnboardingState.ts

```diff
@@ -49,7 +49,7 @@ export function useOnboardingState(): {
 
   // If there are any configured LLM providers already present, skip to the final step
   useEffect(() => {
-    if (Array.isArray(llmProviders) && llmProviders.length > 0) {
+    if (llmProviders) {
```

Comment:
Ooh, I think this is a bug. Try running the following in a `node` shell:

```ts
> if ([]) console.log(0)
```

It prints out `0`, which implies that `[]` is truthy. I believe the logic you want is that `llmProviders?.length > 1`, which is technically different than what you have here.